### PR TITLE
use auth cookies to avoid switch max sessions failure

### PR DESCRIPTION
**Pull request by leslie-qiwa**
Add capability to reuse same session for each request. ref https://www.cisco.com/c/en/us/td/docs/switches/datacenter/nexus9000/sw/6-x/programmability/guide/b_Cisco_Nexus_9000_Series_NX-OS_Programmability_Guide/b_Cisco_Nexus_9000_Series_NX-OS_Programmability_Guide_chapter_011.html

NX-API provides a session-based cookie, nxapi_auth when users first
successfully authenticate. With the session cookie, the username and
password are included in all subsequent NX-API requests that are sent to
the device. The username and password are used with the session cookie
to bypass performing the full authentication process again. If the
session cookie is not included with subsequent requests, another session
cookie is required and is provided by the authentication process.
Avoiding unnecessary use of the authentication process helps to reduce
the workload on the device.

Signed-off-by: Leslie Qi Wang <qiwa@pensando.io>

### CODE REVIEW

**Comment by leslie-qiwa** [comment]
Before this patch, everytime, new sessions are created, for example,
below are some HTTP responses:
```
map[Anticsrf:[V1n/CepQQl8wAcBbvwasJg+NBP0=] Connection:[keep-alive] Content-Security-Policy:[block-all-mixed-content; base-uri 'self'; default-src 'self'; script-src 'self' 'nonce-AQUxi7VoSgRiIa8IOjT GCcN0RYkd0Oub'; style-src 'self' 'nonce-AQUxi7VoSgRiIa8IOjTGCcN0RYkd0Oub'; img-src 'self' data:; connect-src 'self'; font-src 'self'; object-src 'none'; media-src 'self'; form-action 'self'; frame-ancesto rs 'self';] Content-Type:[application/json-rpc; charset=UTF-8] Date:[Wed, 02 Feb 2022 18:52:13 GMT] Server:[nginx/1.7.10] Set-Cookie:[nxapi_auth=dzqnf:XpZdw1I4aFsdeofWDZW37zdnV+Y=; Secure; HttpOnly;] Strict-Transport-Security:[max-age=31536000; includeSubDomains] X-Content-Type-Options:[nosniff] X-Frame-Options:[SAMEORIGIN]]
map[Anticsrf:[zafeFwh4xOUgnfxa/CCPQxfhFhc=] Connection:[keep-alive] Content-Security-Policy:[block-all-mixed-content; base-uri 'self'; default-src 'self'; script-src 'self' 'nonce-0dARyOknMLsC9L5wwFv 8w069h4TcoFtr'; style-src 'self' 'nonce-0dARyOknMLsC9L5wwFv8w069h4TcoFtr'; img-src 'self' data:; connect-src 'self'; font-src 'self'; object-src 'none'; media-src 'self'; form-action 'self'; frame-ancesto rs 'self';] Content-Type:[application/json-rpc; charset=UTF-8] Date:[Wed, 02 Feb 2022 18:52:20 GMT] Server:[nginx/1.7.10] Set-Cookie:[nxapi_auth=dzqnf:kFoo/GfJhNeLrNCarKU5cRtMnNk=; Secure; HttpOnly;] Strict-Transport-Security:[max-age=31536000; includeSubDomains] X-Content-Type-Options:[nosniff] X-Frame-Options:[SAMEORIGIN]]
map[Anticsrf:[uLJ5yBfqehj6pC53NSuFXlRQzQU=] Connection:[keep-alive] Content-Security-Policy:[block-all-mixed-content; base-uri 'self'; default-src 'self'; script-src 'self' 'nonce-X96gOTuG3xY4ECDkPOd yQVw5gygXxoRd'; style-src 'self' 'nonce-X96gOTuG3xY4ECDkPOdyQVw5gygXxoRd'; img-src 'self' data:; connect-src 'self'; font-src 'self'; object-src 'none'; media-src 'self'; form-action 'self'; frame-ancesto rs 'self';] Content-Type:[application/json-rpc; charset=UTF-8] Date:[Wed, 02 Feb 2022 18:52:22 GMT] Server:[nginx/1.7.10] Set-Cookie:[nxapi_auth=dzqnf:X5Is6N6tvMfm6y3A+yjMmOCl5Vc=; Secure; HttpOnly;] Strict-Transport-Security:[max-age=31536000; includeSubDomains] X-Content-Type-Options:[nosniff] X-Frame-Options:[SAMEORIGIN]]
```

After the change, cookies are only shown at the first response, but
won't show up in the succeeding response
```
map[Anticsrf:[K3nw3TN1hX5IeIYTLBSP3hXi3oc=] Connection:[keep-alive] Content-Security-Policy:[block-all-mixed-content; base-uri 'self'; default-src 'self'; script-src 'self' 'nonce-16iWmwef9jyxThEuj4H CNQdnQufFvFHB'; style-src 'self' 'nonce-16iWmwef9jyxThEuj4HCNQdnQufFvFHB'; img-src 'self' data:; connect-src 'self'; font-src 'self'; object-src 'none'; media-src 'self'; form-action 'self'; frame-ancesto rs 'self';] Content-Type:[application/json-rpc; charset=UTF-8] Date:[Wed, 02 Feb 2022 19:01:57 GMT] Server:[nginx/1.7.10] Set-Cookie:[nxapi_auth=dzqnf:C61ZCcw2YVezrEIT7k/sJ1nzssc=; Secure; HttpOnly;] Strict-Transport-Security:[max-age=31536000; includeSubDomains] X-Content-Type-Options:[nosniff] X-Frame-Options:[SAMEORIGIN]]
map[Anticsrf:[K3nw3TN1hX5IeIYTLBSP3hXi3oc=] Connection:[keep-alive] Content-Security-Policy:[block-all-mixed-content; base-uri 'self'; default-src 'self'; script-src 'self' 'nonce-7M5Vm6ijwNpjTxwC6sn U7zGF1MduSedi'; style-src 'self' 'nonce-7M5Vm6ijwNpjTxwC6snU7zGF1MduSedi'; img-src 'self' data:; connect-src 'self'; font-src 'self'; object-src 'none'; media-src 'self'; form-action 'self'; frame-ancesto rs 'self';] Content-Type:[application/json-rpc; charset=UTF-8] Date:[Wed, 02 Feb 2022 19:01:59 GMT] Server:[nginx/1.7.10] Strict-Transport-Security:[max-age=31536000; includeSubDomains] X-Content-Type-Options:[nosniff] X-Frame-Options:[SAMEORIGIN]]
map[Anticsrf:[K3nw3TN1hX5IeIYTLBSP3hXi3oc=] Connection:[keep-alive] Content-Security-Policy:[block-all-mixed-content; base-uri 'self'; default-src 'self'; script-src 'self' 'nonce-ctjIn68UQTRgz1Jm4Pi JJ3UMyboQahWi'; style-src 'self' 'nonce-ctjIn68UQTRgz1Jm4PiJJ3UMyboQahWi'; img-src 'self' data:; connect-src 'self'; font-src 'self'; object-src 'none'; media-src 'self'; form-action 'self'; frame-ancesto rs 'self';] Content-Type:[application/json-rpc; charset=UTF-8] Date:[Wed, 02 Feb 2022 19:02:01 GMT] Server:[nginx/1.7.10] Strict-Transport-Security:[max-age=31536000; includeSubDomains] X-Content-Type-Options:[nosniff] X-Frame-Options:[SAMEORIGIN]]
```

**Review by greenpau — COMMENTED** [review]
@leslie-qiwa , thank you for the PR! 👍 

Please remove URL and the sample HTTP responses from the commit message. Basically, please make it shorter.

File: pkg/client/client.go [hunk 1]
```diff
@@ -205,6 +212,13 @@ func (cli *Client) SetSecure() error {
 	return nil
 }
 
+// UseCookies indicates session based authentication approach and cookies are used to avoid creating new sessions
```
Comment (greenpau) [inline pkg/client/client.go:215]: @leslie-qiwa , please remove the long description 1-2 lines up to 120 char each is enough.

**Review by greenpau — CHANGES_REQUESTED** [review]
Please review my comments.

**Comment by leslie-qiwa** [comment]
> Please review my comments.

done

File: pkg/client/client.go [hunk 2]
```diff
@@ -205,6 +212,13 @@ func (cli *Client) SetSecure() error {
...
+// https://www.cisco.com/c/en/us/td/docs/switches/datacenter/nexus9000/sw/6-x/programmability/guide/b_Cisco_Nexus_9000_Series_NX-OS_Programmability_Guide/b_Cisco_Nexus_9000_Series_NX-OS_Programmability_Guide_chapter_011.html
```
Comment (greenpau) [inline pkg/client/client.go:216]: @leslie-qiwa , this is too long. Just say `UseCookies indicates session based authentication`

**Review by greenpau — CHANGES_REQUESTED** [review]
@leslie-qiwa , please amend.

File: pkg/client/client.go (hunk 2 above)
Comment (leslie-qiwa) [reply pkg/client/client.go]: done. removed.

**Review by greenpau — APPROVED** [review]

**Pull request merged**